miio/integrations/vacuum/roborock/vacuum.py

```diff
@@ -141,6 +143,33 @@ def __init__(
         self.manual_seqnum = -1
         self._maps: Optional[MapList] = None
         self._map_enum_cache = None
+        self._status_helper = UpdateHelper(self.vacuum_status)
+        self._status_helper.add_update_method("consumables", self.consumable_status)
+        self._status_helper.add_update_method("dnd_status", self.dnd_status)
+        self._status_helper.add_update_method("clean_history", self.clean_history)
+        self._status_helper.add_update_method("last_clean", self.last_clean_details)
+        self._status_helper.add_update_method("mop_dryer", self.mop_dryer_settings)
+
+    def send(
```

Comment:
Why are we overwriting the send method?
We can just add the "unknown_mehtod" response check with raising the exception to the main send method used for all devices right?

Reply:
It's overwritten as that response is roborock-specific, the regular errors are reported inside `error` key of the response payload.

On the multiple status containers, we'll have to definitely think how to do that simpler than what the PR you linked does, it feels currently very complex and potentially error-prone.

Reply:
Alright, than I think this is good to go.

Reply:
Great, I'll fix the tests later and get this merged then! I had an idea about potential solution for the multimaps issue(s). How about just adding a back-reference to the child container during embedding? This way the embedded container can define the descriptors while leveraging information from the parent container. Would that solve at least some of the issues you were solving with passing the containers around?

Reply:
Yes I think that will solve some of the issues.
But we schould make a method/attribute in order to acces the parent/a diffrent embeded container from a embeded container.

Reply:
> Design-wise it would be really clean when one status container corresponds to a response from one method, and while that might not be always doable, I think we should strive towards that where possible.

I can look into that, first I want the caching merged and then I will start working on the MultiMap PR.

Reply:
Are you going to add the back-reference to the child container in this PR, or schould I do that in a seperate PR?

Reply:
I removed the need to pass around the MapList container to the main status container in this commit: https://github.com/rytilahti/python-miio/pull/1614/commits/9161737b1aa07f49b6548ce02ab8cca4087a3215

If you implement the back-refrence such that a child container can acces the main container and through that also other containers I can also remove the need to pass MapList to the CleaningDetails

Reply:
Feel free to create a PR to add the back reference. Otherwise I'll take a look into that later this week.

Reply:
Created #1711